data/examples/shaders/filter/rounded-rect.effect

```diff
@@ -0,0 +1,85 @@
+#define IS_FILTER
+#include "../base.effect"
+
+//------------------------------------------------------------------------------
+// Uniforms
+//------------------------------------------------------------------------------
+
+uniform int _0_corner_radius<
+	string name = "Corner Radius";
+	string suffix = " px";
+	string field_type = "input";
+	int minimum = 0;
+	int maximum = 2000;
+	int step = 10;
+	int scale = 1;
+> = 40;
+
+uniform float4 _1_border<
+	string name = "Border Color";
+	string field_type = "slider";
+	float4 minimum = {0., 0., 0., 0.};
+	float4 maximum = {100., 100., 100., 100.};
+	float4 step = {0.01, 0.01, 0.01, 0.01};
+	float4 scale = {0.01, 0.01, 0.01, 0.01};
+> = {0., 0., 0., 100.};
+
+uniform int _2_border_thickness<
+	string name = "Border Thickness";
+	string suffix = " px";
+	string field_type = "input";
+	int minimum = 0;
+	int maximum = 300;
+	int step = 1;
+	int scale = 1;
+> = 5;
+
+
+//------------------------------------------------------------------------------
+// Technique: Combined Curvature
+//------------------------------------------------------------------------------
+
+float4 CombinedCurvature(VertexInformation vtx) : TARGET {
+    float2 uv = vtx.texcoord0.xy;
+    int corner_radius = _0_corner_radius;
+
+    float2 mirrored_tex_coord = float2(0.5, 0.5) - abs(uv - float2(0.5, 0.5));
+	float4 rgb = InputA.Sample(LinearClampSampler, uv);
+
+    float2 pixel_position = float2(mirrored_tex_coord.x * ViewSize.x, mirrored_tex_coord.y * ViewSize.y);
+    float pixel_distance_from_center = distance(pixel_position, float2(corner_radius, corner_radius));
+    
+    bool is_in_corner = pixel_position.x < corner_radius && pixel_position.y < corner_radius;
+    bool is_within_radius = pixel_distance_from_center <= corner_radius;
+    
+    bool is_within_edge_border = !is_in_corner && (pixel_position.x <= _2_border_thickness || pixel_position.y <= _2_border_thickness);
+    bool is_within_corner_border = is_in_corner && pixel_distance_from_center < corner_radius && pixel_distance_from_center > (corner_radius - _2_border_thickness);
```

Comment:
It is fine to && and || statements which are a single word, but for multiple words or even comparison, the statement should be surrounded by (). This prevents situations where one Driver compiler considers it to be `(a > b) && c` and the other `a > (b && c)` because of a driver bug. 

##### Example:
```
bool bad = d || a > b && c < d;
bool ambigious = d || (a > b) && (c < d);
bool good = d || ((a > b) && (c < d));
```

Please adjust this for every instance.